### PR TITLE
dnsdist: Fail if we can't check the configuration file

### LINKED ISSUE #7611: dnsdist: --config-check returns returncode 0 for non-existing config file
- Program: dnsdist 
 - Issue type: Bug report

### Short description

dnsdist returns 0 as result for a config check of an non-existing file. 

I understand that an empty or nonexistent config file is syntactically valid for DNSDIST, but this is still a little bit surprising. 
My expectation was that explicitly checking a file which cannot be read would result in a non-zero return code.

 - Operating system: CentOS
 - Software version: dnsdist 1.3.3
 - Software source: PowerDNS repository

### Steps to reproduce

`dnsdist --check --config nonexistentfile; echo $?`

### Expected behaviour

Expectation: non-zero return code (because of "unable to read configuration from 'nonexistentfile').

### Actual behaviour

```
$ dnsdist --check --config nonexistentfile; echo $?
Unable to read configuration from 'nonexistentfile'
Configuration 'nonexistentfile' OK!
0
```
Note the ZERO returncode.

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Closes #7611.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/docs/upgrade_guide.rst [hunk 1]
```diff
@@ -8,6 +8,8 @@ Responses to AXFR and IXFR queries are no longer cached.
 
 Cache-hits are now counted as responses in our metrics.
 
+The configuration check will now fails if the configuration file does not exist. For this reason we now create a default configuration file, based on the file previously called dnsdistconf.lua, which contains commented-out examples of how to set up dnsdist.
```
Comment (Habbie) [inline pdns/dnsdistdist/docs/upgrade_guide.rst:11]:
```suggestion
The configuration check will now fail if the configuration file does not exist. For this reason we now create a default configuration file, based on the file previously called dnsdistconf.lua, which contains commented-out examples of how to set up dnsdist.
```

Comment (Habbie) [reply pdns/dnsdistdist/docs/upgrade_guide.rst:11]: and maybe quote (double backticks I think?) `dnsdistconf.lua`

**Review by Habbie — COMMENTED** [review]
minor doc nits, otherwise good

**Review by chbruyand — APPROVED** [review]

**Comment by rgacogne** [comment]
Rebased to fix a conflict.

**Pull request merged**